app/src/main/java/com/example/concordia_campus_guide/Activities/PathsActivity.java

```diff
@@ -0,0 +1,139 @@
+        }
+
+        locationFragment = (LocationFragment) getSupportFragmentManager().findFragmentById(R.id.pathLocationFragment);
+
+        from = mViewModel.getFrom();
+        to = mViewModel.getTo();
+        checkFromToType(from, to);
+
+        fromTextView.setText(from.getDisplayName());
+        toTextView.setText(to.getDisplayName());
+        setBackButtonOnClickListener();
+
+        directionWrappers = !(fromIsIndoor && toIsIndoor) ? (ArrayList<DirectionWrapper>) parseDirectionResults() : new ArrayList<DirectionWrapper>();
+        if (!(fromIsIndoor && toIsIndoor)) {
+            locationFragment.drawOutdoorPaths(directionWrappers);
+        }
+
+        View pathInfoCard = findViewById(R.id.path_info_card);
+        swipeableInfoCard = BottomSheetBehavior.from(pathInfoCard);
+        setIndoorPaths();
+        showInfoCard();
+    }
+
+    private void returnToSelectRoute() {
+        Intent returnToSelectRoute = new Intent(PathsActivity.this,
+                RoutesActivity.class);
+        startActivity(returnToSelectRoute);
+    }
+
+    private void setBackButtonOnClickListener() {
+        backButton = (ImageButton) findViewById(R.id.pathsPageBackButton);
+        backButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                returnToSelectRoute();
+            }
+        });
+    }
+
+    public void showInfoCard() {
+        pathInfoCardFragment = new PathInfoCardFragment();
+        Bundle infoCardBundle = new Bundle();
+        infoCardBundle.putSerializable("directionsResult", directionWrappers);
+        if(fromIsIndoor || toIsIndoor) infoCardBundle.putSerializable("walkingPoints", (ArrayList<WalkingPoint>) locationFragment.getWalkingPointList());
+        pathInfoCardFragment.setArguments(infoCardBundle);
+        fragmentTransaction = fragmentManager.beginTransaction();
+        fragmentTransaction.add(R.id.path_info_card_frame, pathInfoCardFragment);
+        fragmentTransaction.commit();
+        swipeableInfoCard.setState(BottomSheetBehavior.STATE_COLLAPSED);
+    }
+
+    public List<DirectionWrapper> parseDirectionResults() {
+        ArrayList<DirectionWrapper> directionWrapperArrayList = new ArrayList<>();
+        DirectionsStep[] steps = directionsResult.routes[0].legs[0].steps;
+        for (DirectionsStep step : steps) {
+            directionWrapperArrayList.add(new DirectionWrapper(step));
+        }
+        return directionWrapperArrayList;
+    }
+    //checks if from and to are indoor places and stores in a boolean array {from, to}
+    public void checkFromToType(Place from, Place to) {
+        fromIsIndoor = from instanceof RoomModel;
+        toIsIndoor = to instanceof RoomModel;
+    }
+
+    public void setIndoorPaths() {
+        if(!fromIsIndoor && !toIsIndoor) return;
+        String floorCode;
+        Place tempFrom = from;
+        Place tempTo = to;
+
+        if(fromIsIndoor && !toIsIndoor){
```

Comment:
any logic checks if it goes from Vanier Library Room 101 to Hall building room 956?

Reply:
That's a very good catch, we have to fix that. It currently crashes.

Reply:
Why ?